app/addons/documents/views.js

```diff
@@ -564,15 +591,79 @@ function(app, FauxtonAPI, Components, Documents, Databases, Views, QueryOptions)
 
       this.toggleTrash();
       this.maybeHighlightAllButton();
+      this.updateExpandButtonLabel();
     },
 
     perPage: function () {
       return this.allDocsNumber.perPage();
+    },
+
+    // a bit dense, but necessary. It decides on what the Expand/Collapse button label should be and takes into
+    // account the default expand/collapse state, what rows are checked and their states
+    updateExpandButtonLabel: function () {
+
+      if (this.defaultDocsStateExpanded) {
+        this.setExpandButtonLabel(false);
+
+        // override the default label and show "Expand" if ALL checked rows have state === collapse (false)
+        if (this.numSelectedOnPage() && this.allCheckedRowsHaveState(false)) {
+          this.setExpandButtonLabel(true);
+        }
+      } else {
+        this.setExpandButtonLabel(true); // default show "Expand"
+
+        // show "Collapse" if ALL checked rows have state === expanded (true)
+        if (this.numSelectedOnPage() && this.allCheckedRowsHaveState(true)) {
+          this.setExpandButtonLabel(false);
+        }
+      }
+    },
+
+    setExpandButtonLabel: function(expanded) {
+      if (expanded) {
+        this.$('#collapse').html('<i class="icon-plus"></i> Expand');
```

Comment:
i mentioned you in https://github.com/robertkowalski/couchdb-fauxton/commit/d92f759960bbae0014b1f2b24fb9ec827ee05f3b at several places to show you what is going to change